Tools/TestWebKitAPI/Tests/WebKitCocoa/WKContentExtensionStore.mm

```diff
@@ -639,6 +639,212 @@ static void checkURLs(const Vector<String>& actual, const Vector<String>& expect
+                "value": "second-test-value"
+            } ] },
+            "trigger": { "url-filter": "testscheme" }
+        }, {
+            "action": { "type": "modify-headers", "priority": 20, "request-headers": [ {
+                "operation": "set",
+                "header": "test-header",
+                "value": "should-not-apply"
+            } ] },
+            "trigger": { "url-filter": "testscheme" }
+        }, {
+            "action": { "type": "modify-headers", "priority": 40, "request-headers": [ {
+                "operation": "remove",
+                "header": "test-header"
+            } ] },
+            "trigger": { "url-filter": "testscheme" }
+        }, {
+            "action": { "type": "modify-headers", "priority": 80, "request-headers": [ {
+                "operation": "set",
+                "header": "test-header",
+                "value": "test-value"
+            } ] },
+            "trigger": { "url-filter": "testscheme" }
+        }]
+    )JSON");
+
+    __block bool receivedAllRequests { false };
+    auto handler = adoptNS([TestURLSchemeHandler new]);
+    handler.get().startURLSchemeTaskHandler = ^(WKWebView *, id <WKURLSchemeTask> task) {
+        NSString *path = task.request.URL.path;
+        NSDictionary<NSString *, NSString *> *fields = [task.request allHTTPHeaderFields];
+        if ([path isEqualToString:@"/main.html"]) {
+            EXPECT_WK_STREQ(fields[@"test-header"], "test-value; second-test-value");
+            receivedAllRequests = true;
+            return respond(task, "");
+        }
+        ASSERT_NOT_REACHED();
+    };
+
+    auto configuration = adoptNS([WKWebViewConfiguration new]);
+    [[configuration userContentController] addContentRuleList:list.get()];
+    [configuration setURLSchemeHandler:handler.get() forURLScheme:@"testscheme"];
+    auto webView = adoptNS([[WKWebView alloc] initWithFrame:NSZeroRect configuration:configuration.get()]);
+    auto delegate = navigationDelegateAllowingActiveActionsOnTestHost().get();
+    webView.get().navigationDelegate = delegate;
+    __block bool receivedActionNotification { false };
+    __block Vector<String> urls;
+    delegate.contentRuleListPerformedAction = ^(WKWebView *, NSString *identifier, _WKContentRuleListAction *action, NSURL *url) {
+        urls.append(url.absoluteString);
+        EXPECT_TRUE(action.modifiedHeaders);
+        receivedActionNotification = true;
+    };
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"testscheme://testhost/main.html"]]];
+    TestWebKitAPI::Util::run(&receivedAllRequests);
+    TestWebKitAPI::Util::run(&receivedActionNotification);
+    checkURLs(urls, {
+        "testscheme://testhost/main.html"_s,
+    });
+
+    // FIXME: Appending to the User-Agent replaces the user agent because we haven't added the user agent yet when processing the request.
+}
+
+TEST_F(WKContentRuleListStoreTest, ModifyHeadersWithCompetingRulesWhereRemoveWins)
```

Comment:
Can we add a test with multiple rule lists that modify the same header?

Reply:
Misunderstood this comment! Still need to address this.